pkg/istio-agent/health/health_check.go

```diff
@@ -49,6 +50,11 @@ const (
 	lastStateUnhealthy
 )
 
+var (
+	legacyLocalhostProbeDestination = env.RegisterBoolVar("REWRITE_PROBE_LEGACY_LOCALHOST_DESTINATION", false,
```

Comment:
might be nice to just move the existing env var to a common place in pkg/istio-agent. Also with the change suggested below we will probably want to share a few more things around setting up the LocalAddr anyways

Reply:
isn't this already in `pkg/istio-agent`? where else would it go?

Reply:
Then can we use it instead of redefining it here?

Reply:
yeah, I just imported from `pilot-agent`. Didn't do that initially for some reason lol